### PR TITLE
Fix broken build.

### PR DESCRIPTION
**Pull request by hubuk**
Added types used by contracts for Task class.

### CODE REVIEW

**Comment by hubuk** [comment]
The build is not quite fixed yet. It worked locally. Will take a look once more.

**Comment by SergeyTeplyakov** [comment]
That is the reason why I'm not confident with all the contract changes:) They breaks the build constantly.

Should I revert that PR? And ask for tests before merging the next one?

**Comment by hubuk** [comment]
Reverting would not be required. At least from what I was able to diagnose. It seems like a some kind of configuration issue. I would have some more info in one hour or so.
Next PRs would be tested automatically by AppVeyor so we should be on a safer side now.

**Comment by SergeyTeplyakov** [comment]
@hubuk Sounds good. So I'll be working on release (put a list of features that we've implemented and other stuff).

Thanks a lot for doing this!

**Comment by hubuk** [comment]
@SergeyTeplyakov There were some minor issues related to differences in public interface between framework versions. The major problem was in lack of .Net 4.6 reference assemblies in repository. Temporary fix requires one more property in system contract projects which contains path to fallback location - ($(MSBuildProgramFiles32)\Reference Assemblies\Microsoft\Framework). Lets wait for AppVeyor build to verify.

I will raise a separate issue that should consider long term solution.

**Comment by hubuk** [comment]
Build failed due to some packages instability. This runs the same commit id: https://ci.appveyor.com/project/hubuk/codecontracts/build/1.9.10126.154/messages

**Comment by SergeyTeplyakov** [comment]
Scheduled a new build at appveyor and trying to build your branch on my local machine..

**Comment by SergeyTeplyakov** [comment]
My local build failed with this error message:

"d:\Sources\CodeContracts_ST\Microsoft.Research\ManagedContract.Setup\buildMSI1
0.xml" (All target) (1) ->
"d:\Sources\CodeContracts_ST\CodeContracts.sln" (Clean;Rebuild target) (2) ->
"d:\Sources\CodeContracts_ST\Microsoft.Research\Contracts\MsCorlib\MsCorlib.Con
tracts10.csproj.metaproj" (Rebuild target) (41:3) ->
"d:\Sources\CodeContracts_ST\Microsoft.Research\Contracts\MsCorlib\MsCorlib.Con
tracts10.csproj" (Rebuild target) (42:3) ->
(BuildMultiTargets target) ->
  System.Threading.Tasks.Task.cs(1621,43): error CS1061: 'System.Threading.Canc
ellationToken' does not contain a definition for 'IsCancellationRequested' and
no extension method 'IsCancellationRequested' accepting a first argument of typ
e 'System.Threading.CancellationToken' could be found (are you missing a using
directive or an assembly reference?) [d:\Sources\CodeContracts_ST\Microsoft.Res
earch\Contracts\MsCorlib\MsCorlib.Contracts10.csproj]
  System.Threading.Tasks.Task.cs(1628,31): error CS0246: The type or namespace
name 'ArgumentOutOfRangeException' could not be found (are you missing a using
directive or an assembly reference?) [d:\Sources\CodeContracts_ST\Microsoft.Res
earch\Contracts\MsCorlib\MsCorlib.Contracts10.csproj]
  System.Threading.Tasks.Task.cs(1650,43): error CS1061: 'System.Threading.Canc
ellationToken' does not contain a definition for 'IsCancellationRequested' and
no extension method 'IsCancellationRequested' accepting a first argument of typ
e 'System.Threading.CancellationToken' could be found (are you missing a using
directive or an assembly reference?) [d:\Sources\CodeContracts_ST\Microsoft.Res
earch\Contracts\MsCorlib\MsCorlib.Contracts10.csproj]
  System.Threading.Tasks.Task.cs(1657,31): error CS0246: The type or namespace
name 'ArgumentOutOfRangeException' could not be found (are you missing a using
directive or an assembly reference?) [d:\Sources\CodeContracts_ST\Microsoft.Res
earch\Contracts\MsCorlib\MsCorlib.Contracts10.csproj]

**Comment by SergeyTeplyakov** [comment]
To fix this issue please do the following:

1. Put following code at MsCorlib\System.Threading.CancellationToken.cs at line 81:

```csharp
    public bool IsCancellationRequested { get {return false; } }
```

And then comment out two lines in the MsCorLib\System.Threading.Tasks.cs: 

```csharp
Contract.EnsuresOnThrow<System.ArgumentOutOfRangeException>(true, "Cancellation has not been requested for cancellationToken; its IsCancellationRequested property is false.");
```

I've commented them out (there 2 use cases for this and put a comment:

```csharp
      // Commenting out to fix a build break
      // Contract.EnsuresOnThrow<System.ArgumentOutOfRangeException>(true, "Cancellation has not been requested for cancellationToken; its IsCancellationRequested property is false.");

```

File: Microsoft.Research/Contracts/MsCorlib/System.Threading.CancellationToken.cs [hunk 1]
```diff
@@ -78,7 +78,14 @@ public struct CancellationToken
     //
     // Returns:
     //     true if cancellation has been requested for this token; otherwise false.
-    //public bool IsCancellationRequested { get; }
+    public bool IsCancellationRequested
```
Comment (hubuk) [inline Microsoft.Research/Contracts/MsCorlib/System.Threading.CancellationToken.cs]: This is the first place already fixed

File: Microsoft.Research/Contracts/MsCorlib/System.ArgumentOutOfRangeException.cs [hunk 2]
```diff
@@ -12,48 +12,41 @@
...
 
-using System.Diagnostics.Contracts;
 using System;
+using System.Diagnostics.Contracts;
 
 namespace System
 {
 
-  public class ArgumentOutOfRangeException
+  public class ArgumentOutOfRangeException : Exception
```
Comment (hubuk) [inline Microsoft.Research/Contracts/MsCorlib/System.ArgumentOutOfRangeException.cs]: This is the second one.

**Comment by hubuk** [comment]
It seems that both problems are already addressed. Check if you are at latest commit with your local copy.

**Comment by SergeyTeplyakov** [comment]
Hm.. you're right. It seems that I didn't fetch your branch properly.
trying it once more time.

**Comment by SergeyTeplyakov** [comment]
Confirm that the fix is working on my machine. Merging...

**Pull request merged**